testsuite/cluster-test/src/experiments/performance_benchmark.rs

```diff
@@ -160,8 +192,32 @@ impl Experiment for PerformanceBenchmark {
             "Link to dashboard : {}",
             context.prometheus.link_to_dashboard(start, end)
         );
-        let futures = stop_effects.iter().map(|e| e.deactivate());
-        join_all(futures).await;
+        if let Some(cluster_swarm) = context.cluster_swarm {
+            let instance_configs: Vec<&InstanceConfig> = self
+                .down_instances
+                .iter()
+                .map(|instance| -> Result<&InstanceConfig> {
+                    instance
+                        .instance_config()
+                        .ok_or_else(|| format_err!("Failed to find instance_config"))
+                })
+                .collect::<Result<_, _>>()?;
+            let futures: Vec<_> = instance_configs
+                .into_iter()
+                .map(|ic| match ic {
+                    Validator(validator_config) => {
+                        cluster_swarm.upsert_validator(validator_config.clone(), false)
```

Comment:
Similarly, can we have `upsert_node` function in swarm?

Reply:
Done